sema/check/check.py

```
@@ -0,0 +1,133 @@
+    log.info(f"Loaded {len(rules)} rules from {input_folder}")
+    results = []
+    for i, rule in enumerate(rules, 1):
+        try:
+            test = instantiatie_impl_checks(rule)
+            log.debug(
+                f"Running test {i}/{len(rules)}: {test.__class__.__name__}"
+            )
+            result = test.run()
+            results.append(result)
+        except Exception as e:
+            log.exception(f"Error running test {i}/{len(rules)}: {e}")
```

Ruff

Logging statement uses f-string

(G004)